node/pkg/watchers/sui/watcher.go

```diff
@@ -231,189 +249,221 @@ func (e *Watcher) inspectBody(logger *zap.Logger, body SuiResult) error {
 
-	logger.Info("Subscribing using", zap.String("filter", s))
+	logger.Info("Subscribing using", zap.String("json:", subscription))
 
-	if err := ws.WriteMessage(websocket.TextMessage, []byte(s)); err != nil {
-		logger.Error(fmt.Sprintf("write: %s", err.Error()))
-		return err
+	err = ws.Write(ctx, websocket.MessageText, []byte(subscription))
+	if err != nil {
+		p2p.DefaultRegistry.AddErrorCount(vaa.ChainIDSui, 1)
+		suiConnectionErrors.WithLabelValues("websocket_subscription_error").Inc()
+		return fmt.Errorf("websocket subscription failed: %w", err)
+	}
+	// Wait for the success response
+	mType, p, err := ws.Read(ctx)
+	if err != nil {
+		p2p.DefaultRegistry.AddErrorCount(vaa.ChainIDSui, 1)
+		suiConnectionErrors.WithLabelValues("event_subscription_error").Inc()
+		return fmt.Errorf("event subscription failed: %w", err)
+	}
+	var subRes map[string]any
+	err = json.Unmarshal(p, &subRes)
+	if err != nil {
+		return fmt.Errorf("failed to Unmarshal the subscription result: %w", err)
+	}
+	logger.Info("Unmarshalled json", zap.Any("subRes", subRes))
+	actualResult := subRes["result"]
+	logger.Info("actualResult", zap.Any("res", actualResult))
+	if actualResult == nil {
+		return fmt.Errorf("Failed to request filter in subscription request")
 	}
+	logger.Info("subscribed to new transaction events", zap.Int("messageType", int(mType)), zap.String("bytes", string(p)))
 
-	timer := time.NewTicker(time.Second * 1)
+	timer := time.NewTicker(time.Second * 5)
 	defer timer.Stop()
 
-	supervisor.Signal(ctx, supervisor.SignalHealthy)
-
 	errC := make(chan error)
-	defer close(errC)
 	pumpData := make(chan []byte)
 	defer close(pumpData)
 
-	go func() {
+	supervisor.Signal(ctx, supervisor.SignalHealthy)
+	readiness.SetReady(e.readinessSync)
+
+	common.RunWithScissors(ctx, errC, "sui_data_pump", func(ctx context.Context) error {
 		for {
-			if _, msg, err := ws.ReadMessage(); err != nil {
-				logger.Error(fmt.Sprintf("ReadMessage: '%s'", err.Error()))
-				if strings.HasSuffix(err.Error(), "EOF") {
-					errC <- err
-					return
+			select {
+			case <-ctx.Done():
+				logger.Error("sui_data_pump context done")
+				return ctx.Err()
+
+			default:
+				_, msg, err := ws.Read(ctx)
+				if err != nil {
+					logger.Error(fmt.Sprintf("ReadMessage: '%s'", err.Error()))
+					p2p.DefaultRegistry.AddErrorCount(vaa.ChainIDSui, 1)
+					suiConnectionErrors.WithLabelValues("channel_read_error").Inc()
+					return err
+				}
+
+				var res SuiEventMsg
+				err = json.Unmarshal(msg, &res)
+				if err != nil {
+					logger.Error("Failed to unmarshal SuiEventMsg", zap.String("body", string(msg)), zap.Error(err))
+					return fmt.Errorf("Failed to unmarshal SuiEventMsg, body: %s, error: %w", string(msg), err)
+				}
+				if res.Error != nil {
+					return fmt.Errorf("Bad SuiEventMsg, body: %s, error: %w", string(msg), err)
+				}
+				logger.Info("SUI result message", zap.String("message", string(msg)), zap.Any("event", res))
```

Comment:
Need to remember to change these things to debugs before mainnet.

Reply:
Done